### ISSUE TITLE
[ANDROID] Add option for retrofit2

### ISSUE THREAD

**Issue by ashughes** [swagger-api/swagger-codegen#3882]
##### Description

Retrofit 2 is available with the Java language, but not Android.
##### Swagger-codegen version

2.2.1
##### Command line used for generation

java -jar swagger-codegen-cli.jar generate -i swagger.yaml -l java -o generated --library retrofit2 -DuseRxJava=true,hideGenerationTimestamp=true

Currently we're using Java, but we ran into some Android compatibility issues, so it would be ideal to have retrofit2 available for Android.

**Comment by cbornet** [comment]
Can you describe what compatibility issues you faced ? I use the java/retrofit2 gen on android without problem.

**Comment by ashughes** [comment]
@cbornet Sorry, yes, I made a comment on #3568 regarding Android compatibility.

**Comment by njwandroid** [comment]
@cbornet @ashughes  Is there some sort of a quick start guide for retrofit2+swagger+android?  The docs make it seem like I should put something in pom.xml, which doesn't seem to apply, I don't know if I'm better off with using the .jar and importing it or the library, etc, etc.  Or perhaps we could have a .sh file for androidRetrofit, since retrofit is probably *more* popular than Volley?

**Comment by wing328** [comment]
@njwandroid I believe the doc you're referring to is similar to this one: https://github.com/swagger-api/swagger-codegen/blob/master/samples/client/petstore/java/retrofit2/README.md

We need to update it with better installation instruction such as using `gradle`, which is more popular among Android developers.

Java (okhttp-gson) API client has better doc for installation with `gradle`: https://github.com/swagger-api/swagger-codegen/tree/master/samples/client/petstore/java/okhttp-gson#gradle-users. Does it meet your requirement?

**Comment by njwandroid** [comment]
On Wed, Mar 8, 2017 at 7:13 PM, wing328 <notifications@github.com> wrote:

> https://github.com/swagger-api/swagger-codegen/tree/master/samples/client/
> petstore/java/okhttp-gson#gradle-users


Thank you so much for your prompt response.
That is a lot closer and has some awesome stuff in it - *except* it still
doens't look like it will "just work"

Ideally it should be tested to work step by step by someone that's never
used maven by commandline or used swagger before.

e.g. compile "io.swagger:swagger-petstore-okhttp-gson:1.0.0" won't "just
work" will it?  Are you saying that will work after running mvn install -
and if you have the m2 home variable set or something?

If so, that means it won't work for Jenkins and/or won't work if you check
out the repo fresh right?

That's a big item to note.

I decided after working on it a good while to just use mvn package to make
a jar, put the jar in the project.  It was a plus to including the source
code - which definitely did not "just work" - at least with version 2.2.1.

The process was confusing enough for me that I gave up until I was required
to return to it.

Now I'm happy I found a way and I'm happy you responded promptly, but it
should have steps that can be followed step by step.

Thanks!
<https://github.com/swagger-api/swagger-codegen/tree/master/samples/client/petstore/java/okhttp-gson#others>
ᐧ
